src/game/client/ingameui.h

```diff
@@ -218,16 +395,92 @@ class InGameUI : public SubsystemInterface, public SnapShot
+    void Add_Named_Timer(const Utf8String &timer_name, const Utf16String &text, bool countdown);
+    void Remove_Named_Timer(const Utf8String &timer_name);
+    void Show_Named_Timer_Display(bool show);
+    void Handle_Radius_Cursor();
+    void Trigger_Double_Click_Attack_Move_Guard_Hint();
+    void Evaluate_Solo_Nexus(Drawable *drawable);
+    void Handle_Build_Placements();
+    void Register_Window_Layout(WindowLayout *layout);
+    void Unregister_Window_Layout(WindowLayout *layout);
+    void Free_Message_Resources();
+    void Add_Message_Text(const Utf16String &formatted_message, const RGBColor *rgb_color);
+    void Remove_Message_At_Index(int i);
+    void Destroy_Placement_Icons();
+    void Expire_Hint(HintType type, unsigned int hint_index);
+    void Create_Control_Bar();
+    void Create_Replay_Control();
+    bool Are_Selected_Objects_Controllable() const;
+    void Reset_Camera();
+    bool Can_Selected_Objects_Non_Attack_Interact_With_Object(const Object *object_to_interact_with, SelectionRules rule);
+    CanAttackResult Get_Can_Selected_Objects_Attack(
+        ActionType action, const Object *object_to_interact_with, SelectionRules rule, bool force_to_attack) const;
+    bool Can_Selected_Objects_Do_Action(
+        ActionType action, const Object *object_to_interact_with, SelectionRules rule, bool additional_checking) const;
+    bool Can_Selected_Objects_Do_Special_Power(
+        CommandButton *command, const Object *object_to_interact_with, Coord3D *position, SelectionRules rule) const;
+    bool Can_Selected_Objects_Override_Special_Power_Destination(
+        const Coord3D *pos, SelectionRules rule, SpecialPowerType type) const;
+    bool Can_Selected_Objects_Effectively_Use_Weapon(
+        CommandButton *command, const Object *object_to_interact_with, Coord3D *position, SelectionRules rule) const;
+    void Update_Floating_Text();
+    void Draw_Floating_Text();
+    void Clear_Floating_Text();
+    void Clear_Popup_Message_Data();
+    void Clear_World_Animations();
+    void Update_And_Draw_World_Animations();
+
+    static void Parse_In_Game_UI_Definition(INI *ini);
 
     bool Get_Drawable_Caption_Bold() const { return m_drawableCaptionBold; }
     int Get_Drawable_Caption_Size() const { return m_drawableCaptionPointSize; }
     Utf8String Get_Drawable_Caption_Font() const { return m_drawableCaptionFont; }
     int Get_Drawable_Caption_Color() const { return m_drawableCaptionColor; }
-    bool Is_In_Waypoint_Mode() const { return m_waypointMode; }
-    bool Get_Unk7() const { return m_unk7; }
-    void Set_Unk6(bool b) { m_unk6 = b; }
+    bool Get_Draw_RMB_Scroll_Anchor() const { return m_drawRMBScrollAnchor; }
+    bool Get_Move_RMB_Scroll_Anchor() const { return m_moveRMBScrollAnchor; }
+    const PopupMessageData *Get_Popup_Message_Data() const { return m_popupMessageData; }
+
+    int Get_Message_Color(bool color) const

```

Comment:
`bool secondary`
or
`bool secondary = false`
perhaps.

In INI they look like so:
```
  MessageColor1                  = R:255 G:255 B:255
  MessageColor2                  = R:180 G:180 B:255
```

Reply:
fixed.